docs/source/configuration/cors.mdx

```diff
@@ -1,13 +1,143 @@
+
+If you instead rely on `allow_any_origin`, while your personal computer is on your private network, a script on any website could potentially make your browser talk to your private API. [Some browsers, such as Chrome, have features under development](https://wicg.github.io/private-network-access/) to solve this problem. But in the meantime, all servers on private networks should *always* specify origins in their CORS configuration.
+
+#### APIs that require cookies
+
+If your API needs to accept [cross-origin cookies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies) with requests, you _must_ specify origins in your CORS configuration. Otherwise, cross-origin cookies are automatically disabled. This is _not_ a security vulnerability, but it does prevent your API from successfully providing cookies.
+
+For examples, see [Passing credentials with CORS](#passing-credentials-with-cors).
+
+#### APIs with known consumers
+
+If you create an API on the public internet to serve resources to your _own_ websites or apps, you want to [specify which origins](#configuring-cors-options-for-apollo-server) can access your server's resources.
+
+#### Public or embedded APIs
+
+If you create a public API or an API to embed in websites you don't control yourself, you probably want to allow _any_ origin to access your server's resources. You can set `allow_any_origin` in your `router.yml` configuration:
+
+```yml
+server:
+  cors:
+    allow_any_origin: true
+```
+
+> Using `allow_any_origin` makes the router send the [wildcard (`*`)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#directives) value for the ACAO header, which enables _any_ website to tell a user's browser to send an arbitrary request (without cookies or other credentials) to your server and read that server's response.
+
+#### APIs with non browser consumers
+
+The default Apollo Server CORS behavior should suit your use case, and https://studio.apollographql.com/sandbox/explorer allows you to run queries against your Apollo Router.
+You can always specify origins in your CORS configuration later. 
+
+##  Configuring CORS options for Apollo Server
+
+By default, the apollo router only allows requests from https://studio.apollographql.com. If your Apollo Router is serving data for web applications, you want to either add their origin to the list of allowed origins, or switch to `allow_any_origin`.
+
+> If you need to pass credentials to your server (e.g., via cookies), you can't use the `allow_any_origin` flag. You _must_ provide specific origins. For examples, see [Passing credentials with CORS](#passing-credentials-with-cors).
+
+The batteries-included `apollo-server` [has a `cors` option](../api/apollo-server/#cors), which you can use to change your server's default CORS behavior:
```

Comment:
`[has a `cors` option](../api/apollo-server/#cors)` is a 404 in the preview. Maybe use the absolute URL.

Reply:
Good catch! this should have been removed!

Reply:
fixed in 71bf26c8